fredericia/web
# FKDK-945 File transfer WS callbacks.

## This PR will be merged when customer will approve the changes on testing env

https://os2web.atlassian.net/browse/FKDK-945

Related PR https://github.com/bellcom/os2web_profile/pull/9

## Required software and access:
- SOAPIU 
- DB administration tool
- Access to server

## Preparation:
- Install SOAP UI tool and create new project
- Specify URL http://test.fredericia.bellcom.dk/os2web/esdh/om/v1?wsdl as source to WS
In the end of setup you should get next WS in SOAP IU ![screenshot from 2017-12-22 09-12-06](https://user-images.githubusercontent.com/5544994/34292356-e65030ea-e700-11e7-8d0a-0810a4ac3687.png)

## Test case ReturnFileTransferRequest
1. In SOAP UI open ReturnFileTransferRequest and submit the request (Press play button in the left top corner)
   No changes in the request xml-body
2. As response you have to get next XML content ![screenshot from 2017-12-22 09-15-42](https://user-images.githubusercontent.com/5544994/34292391-01a488aa-e701-11e7-98b9-a737e29934cb.png)
3. Open drupal DB and get node by responsed file_id using this SQL-request https://gist.github.com/andriyun/1bab94bc256de3a17e4918c2b4552df6
4. Match nid with list of latest non delivered CP Document nodes. Get the list by SQL-request https://gist.github.com/andriyun/f5a8dcbec1ddd94417104d1d0f8ab4ba
5. Open CP Document node and check Delivered status check box.
6. Make request from step 1 and check the response
 -- You should another file_id
7. Restore changes from step 5 and make request from step 1
 -- You should previous file_id
8. Make steps 5,6 with Non transferable checkbox

## Test FileTransferEnd methos
### Case "failed transfer status"
0. Get latest FileID using ReturnFileTransferRequest method.
1. In SOAP UI open FileTransferEnd and set
 -- FileID -> FileID from previous test case
 -- TransferCompletedWithoutErrors -> 0
2. Submit the request (Press play button in the left top corner)
3. Open CP Document node and check fields
 -- Synchronization fails -> should be insreased +1
 -- Transfer status -> should has 'failed transfer status' value
4. Go to spet 2 and make 4-5 requests
5. Open CP Document node and check fields
 -- Synchronization fails -> should be insreased +n times
 -- Non transferable -> should be checked

### Case "transfered file not found"
0. Get latest FileID using ReturnFileTransferRequest method.
1. In SOAP UI open FileTransferEnd and set
 -- FileID -> FileID from previous test case
 -- TransferCompletedWithoutErrors -> 1
2. Submit the request (Press play button in the left top corner)
3. Open CP Document node and check fields
 -- Synchronization fails -> should be insreased +1
 -- Transfer status -> should has 'transfered file not found' value

### Case "Binding delivered file to CP Document node"
0. Get latest FileID using ReturnFileTransferRequest method.
1. Put any pdf file(not the same as CP Document has by PULL download) to path specified in settings /admin/config/os2web/settings ![screenshot from 2017-12-22 09-51-07](https://user-images.githubusercontent.com/5544994/34292440-3be0cad8-e701-11e7-8c57-b96abe1b1afe.png)
 -- set the file name as **[file_id].pdf**
2. In SOAP UI open FileTransferEnd and set
 -- FileID -> FileID from previous test case
 -- TransferCompletedWithoutErrors -> 1
3. Submit the request (Press play button in the left top corner)
4. Open CP Document node and check Indhold - File field
 -- You should have binded file to file field
 -- Synchronization fails -> reseted to 0
 -- Delivered -> should be checked

### Case "File get" functionality.
NOTE: This case require successful case "Binding delivered file to CP Document node".
0. Find the CP Document node with binded file (It can be nid from previous test case) and fetch file_id.
1. Put settings to PULL mode in settings /admin/config/os2web/settings (screenshot)
2. On new browser tab open url /os2web/service/gf/v1/[file_id_from_step_0]
 -- You should get original donwloaded file from OM.
3. Change setting to PUSH and do step 2 again
 -- You should be redirected to local stored file binded to node.
4. Open CP Document node, remove the file and do step 2 again
 -- You should get an error.

Case "Combined mode" to get the file
0. Get any CP Documetn node without binded file and fetch file_id
1. Change setting to Combined mode in settings /admin/config/os2web/settings
2. On new browser tab open url /os2web/service/gf/v1/[file_id_from_step_0]
 -- You should get original donwloaded file from OM.
3. Open CP Document node and check Indhold - File field
 -- You should have binded file to file field
 -- Synchronization fails -> reseted to 0
 -- Delivered -> should be checked
 -- Transfer status -> should has 'combined mode transfer' value
